deployment/cre/ocr3/v2/changeset/operations/contracts/configure_ocr3.go

```diff
@@ -0,0 +1,162 @@
+package contracts
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/mcms"
+	"github.com/smartcontractkit/mcms/sdk"
+
+	mcmstypes "github.com/smartcontractkit/mcms/types"
+
+	cldf "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+	ocr3_capability "github.com/smartcontractkit/chainlink-evm/gethwrappers/keystone/generated/ocr3_capability_1_0_0"
+	capabilities_registry_v2 "github.com/smartcontractkit/chainlink-evm/gethwrappers/workflow/generated/capabilities_registry_wrapper_v2"
+
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/deployment/cre/contracts"
+	"github.com/smartcontractkit/chainlink/deployment/cre/ocr3"
+)
+
+type ConfigureOCR3Deps struct {
+	Env                  *cldf.Environment
+	WriteGeneratedConfig io.Writer
+	Registry             *capabilities_registry_v2.CapabilitiesRegistry
+}
+
+type ConfigureOCR3Input struct {
+	ContractAddress  *common.Address
+	RegistryChainSel uint64
+	DONs             []ConfigureCREDON
+	Config           *ocr3.OracleConfig
+	DryRun           bool
+
+	MCMSConfig *ocr3.MCMSConfig
+}
+
+func (i ConfigureOCR3Input) UseMCMS() bool {
+	return i.MCMSConfig != nil
+}
+
+type ConfigureOCR3OpOutput struct {
+	MCMSTimelockProposals []mcms.TimelockProposal
+}
+
+var ConfigureOCR3 = operations.NewOperation[ConfigureOCR3Input, ConfigureOCR3OpOutput, ConfigureOCR3Deps](
+	"configure-ocr3-op",
+	semver.MustParse("1.0.0"),
+	"Configure OCR3 Contract",
+	func(b operations.Bundle, deps ConfigureOCR3Deps, input ConfigureOCR3Input) (ConfigureOCR3OpOutput, error) {
+		if input.ContractAddress == nil {
+			return ConfigureOCR3OpOutput{}, errors.New("ContractAddress is required")
+		}
+
+		var nodeIDs []string
+		for _, don := range input.DONs {
+			donConfig := RegisteredDonConfig{
+				NodeIDs:          don.NodeIDs,
+				Name:             don.Name,
+				RegistryChainSel: input.RegistryChainSel,
+				Registry:         deps.Registry,
+			}
+			d, err := newRegisteredDon(*deps.Env, donConfig)
+			if err != nil {
+				return ConfigureOCR3OpOutput{}, fmt.Errorf("configure-ocr3-op failed: failed to create registered DON %s: %w", don.Name, err)
+			}
+
+			// We double-check that the DON accepts workflows...
+			if d.Info.AcceptsWorkflows {
+				for _, node := range d.Nodes {
+					nodeIDs = append(nodeIDs, node.NodeID)
+				}
+			}
+		}
+
+		chain, ok := deps.Env.BlockChains.EVMChains()[input.RegistryChainSel]
+		if !ok {
+			return ConfigureOCR3OpOutput{}, fmt.Errorf("chain %d not found in environment", input.RegistryChainSel)
+		}
+
+		if input.ContractAddress == nil {
```

Comment:
We are already performing this check at the beginning of this func. So might just delete this one, and update the first's error message to match this message.